### PR TITLE
add MatrixServerTable filler

### PR DESCRIPTION
**Pull request by xuehui1991**
add a filler in MatrixServerTable init function.

### CODE REVIEW

**Comment by guolinke** [comment]
Some suggestions about "void(*filler)(std::vector<T> &)": 
1. If you need change this parameter in this function, use " std::vector<T>* " instead.
2. Pointer to function and pointer to member function are not the same type in c++. So you cannot pass pointer to member function here. I think it is better by using std::function.

File: include/multiverso/table/matrix_table.h [hunk 1]
```diff
@@ -13,6 +13,7 @@ template <typename T>
 class MatrixWorkerTable : public WorkerTable {
 public:
   MatrixWorkerTable(integer_t num_row, integer_t num_col);
+  MatrixServerTable(integer_t num_row, integer_t num_col, void(*filler)(std::vector<T> &));
```
Comment (chivee) [inline include/multiverso/table/matrix_table.h]: why declare this in the MatrixWorkerTable scope?

File: src/table/matrix_table.cpp [hunk 2]
```diff
@@ -243,6 +243,14 @@ MatrixServerTable<T>::MatrixServerTable(integer_t num_row, integer_t num_col) :
 }
 
 template <typename T>
+MatrixServerTable<T>::MatrixServerTable(integer_t num_row, integer_t num_col, void(*filler)(std::vector<T> &)) :
+MatrixServerTable<T>::MatrixServerTable(num_row, num_col) {
```
Comment (chivee) [inline src/table/matrix_table.cpp]: does this filler function are used to initialized parameter/?

**Comment by feiga** [comment]
I don't think it's a good idea.

**Comment by feiga** [comment]
Close this PR currently. We may use some better way to initialize. @xuehui1991

**Pull request closed without merging**